libfreerdp/core/codecs.c

```diff
@@ -100,14 +100,10 @@ BOOL freerdp_client_codecs_prepare(rdpCodecs* codecs, UINT32 flags)
 	return TRUE;
 }
 
-BOOL freerdp_client_codecs_reset(rdpCodecs* codecs, UINT32 flags,
-				 UINT32 width, UINT32 height)
+BOOL freerdp_client_codecs_reset(rdpCodecs* codecs, UINT32 flags, UINT32 width, UINT32 height)
 {
 	BOOL rc = TRUE;
 
-	if (!freerdp_client_codecs_prepare(codecs, flags))
```

Comment:
The codec context should be indeed only created when it is really used. Reset codec should not call prepare as it should only 'reset' the 'existing' codec. Instead, prepare should invoke codec reset subroutines so that we complete the initialization (Although currently only h264 codec really need reset when it is created, we should anyway keep all codec consistent in the semantic level)

I drafted PR #3376 for fix